compiler/src/dotty/tools/dotc/core/Contexts.scala

```diff
@@ -271,6 +271,18 @@ object Contexts {
       if (this.settings.YtraceContextCreation.value)
         creationTrace = (new Throwable).getStackTrace().take(20)
 
+    /** Whether current prettyprint is for -Xprint
+     *  -Xprint will print `sym.name` instead of `sym.originalName`
+     */
+    private var _isXprintMode = false
```

Comment:
This is far too heavyweight. You are adding a field to every context which will have to copied millions of times. There are more lightweight options: 

 - Add a bit to Mode
 - Add a property 

I think adding a property is preferred as it adds the least footprint and the creation and access is not performance critical (adding and testing a mode bit is faster than adding and testing a property).

Also, the logic turns Contexts into mutable objects, which completely changes their role in the compiler and is at odds about all the assumptions we make about them.